rust-connectors/sinks/postgres/src/connect.rs

```diff
@@ -0,0 +1,373 @@
+            let mut sql_statements: Vec<String> = Vec::new();
+            match event.message {
+                LogicalReplicationMessage::Insert(insert) => {
+                    if let Some(table) = self.relations.get(&insert.rel_id) {
+                        let sql = Self::to_table_insert(table, &insert);
+                        sql_statements.push(sql);
+                    } else {
+                        tracing::error!("Failed to find table for: {:?}", insert);
+                    }
+                }
+                LogicalReplicationMessage::Relation(new_rel) => {
+                    if let Some(old_table) = self.relations.get_mut(&new_rel.rel_id) {
+                        let alters = Self::to_table_alter(&new_rel, old_table);
+                        sql_statements.extend(alters);
+
+                        *old_table = new_rel;
+                    } else {
+                        let sql = Self::to_table_create(&new_rel);
+                        sql_statements.push(sql);
+                        self.relations.insert(new_rel.rel_id, new_rel);
+                    }
+                }
+                LogicalReplicationMessage::Delete(delete) => {
+                    if let Some(table) = self.relations.get(&delete.rel_id) {
+                        let sql = Self::to_delete(table, &delete);
+                        sql_statements.push(sql);
+                    } else {
+                        tracing::error!("Failed to find table for delete: {:?}", delete);
+                    }
+                }
+                LogicalReplicationMessage::Update(update) => {
+                    if let Some(table) = self.relations.get_mut(&update.rel_id) {
+                        let sql = Self::to_update(table, &update);
+                        sql_statements.push(sql);
+                    } else {
+                        tracing::error!("Failed to find table for update: {:?}", update);
+                    }
+                }
+                LogicalReplicationMessage::Truncate(trunk) => {
+                    let sql = Self::to_table_trucate(&self.relations, trunk);
+                    sql_statements.push(sql);
+                }
+                LogicalReplicationMessage::Commit(_) | LogicalReplicationMessage::Begin(_) => {}
+                other => {
+                    tracing::error!("Uncaught replication message: {:?}", other);
+                }
+            }
+            if !sql_statements.is_empty() {
+                let sql = format!(
+                    "UPDATE fluvio.offset SET current_offset={} where id = 1",
+                    offset
+                );
+                sql_statements.push(sql);
+                let batch = sql_statements.join(";");
+                tracing::info!("executing sql: {:?}", batch);
+                self.pg_client.batch_execute(&batch).await?;
+            }
+        }
+        Ok(())
+    }
+
+    pub fn to_table_trucate(
```

Comment:
I think all these `to_table_*`,`to_insert`, etc should go into the `fluvio-postgres-models` crate. There's also some shared stuff from `fluvio-postgres-source` in `convert.rs`.